### PR TITLE
Add more SANs to the master certificate.

### PR DESCRIPTION
**Pull request by roberthbailey**

### CODE REVIEW

**Comment by roberthbailey** [comment]
Should help with https://github.com/GoogleCloudPlatform/kubernetes/issues/7648. /cc @mikedanese

**Comment by roberthbailey** [comment]
I'm going to launch a cluster and manually inspect the certs. Please take a look at the shell but don't merge yet.

**Comment by roberthbailey** [comment]
Here's the cert on the master:

```
$ sudo cat /srv/kubernetes/server.cert 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 1 (0x1)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=107.178.209.103@1434738635
        Validity
            Not Before: Jun 19 18:30:35 2015 GMT
            Not After : Jun 16 18:30:35 2025 GMT
        Subject: CN=kubernetes-master
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
            RSA Public Key: (2048 bit)
                Modulus (2048 bit):
                    00:bd:75:a0:6d:ff:6a:ab:14:eb:6f:49:0b:ff:54:
                    af:ae:26:43:f2:86:fc:b3:57:26:6a:23:5e:98:14:
                    10:c5:47:e4:1c:37:39:a3:79:ed:a0:d4:75:6d:0e:
                    a2:c1:9b:59:4d:8d:fa:a9:d8:50:29:3d:ea:e4:05:
                    e8:66:b6:28:ba:be:2c:98:39:71:1b:19:eb:36:f8:
                    d4:ba:7c:22:9b:a1:9f:e7:5f:b6:09:63:a3:99:65:
                    4f:5e:9e:a9:be:c2:a3:52:04:80:d0:a3:ef:56:2c:
                    ed:1e:4f:c9:9a:b7:58:a2:ba:aa:d7:d2:f2:69:a5:
                    62:ee:4d:07:20:e8:be:c6:50:82:eb:40:b7:da:87:
                    b4:55:9f:39:59:c4:a2:04:e2:92:63:2b:40:56:fe:
                    0c:fd:e1:1f:09:21:4d:ff:69:7d:65:4a:5b:70:d1:
                    99:15:25:44:24:10:6d:c6:5a:66:8b:0a:9a:37:70:
                    28:cd:52:4a:54:cb:11:85:b9:61:21:2f:a1:d8:57:
                    f0:cd:91:05:c2:7b:6e:19:78:27:13:da:44:1c:c5:
                    19:b6:0e:cd:c5:2e:06:7c:b7:75:d6:5b:85:6e:d5:
                    37:cf:0f:25:a1:bb:a5:8b:c9:7b:97:ea:72:de:cf:
                    9e:88:41:64:9d:ae:55:d3:72:ee:f5:c7:57:8b:80:
                    1d:37
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Subject Key Identifier: 
                96:00:1D:4D:74:2A:12:76:E4:F1:95:74:24:5C:CB:82:71:EB:EC:F1
            X509v3 Authority Key Identifier: 
                keyid:ED:D1:25:2A:6E:70:7B:97:E1:15:02:D8:16:7C:AE:FA:BA:13:7C:00
                DirName:/CN=107.178.209.103@1434738635
                serial:F9:AF:1F:9F:71:09:38:B1

            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Key Usage: 
                Digital Signature, Key Encipherment
            X509v3 Subject Alternative Name: 
                IP Address:107.178.209.103, IP Address:10.0.0.1, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:kubernetes-master
    Signature Algorithm: sha256WithRSAEncryption
        b8:0b:87:d8:7b:77:65:d8:d7:99:89:de:f3:9e:7d:29:0f:f6:
        57:db:a3:1f:49:1f:42:37:ef:8c:79:4b:38:c4:cb:20:d5:c8:
        97:5d:ae:3f:9c:d0:94:3c:90:f0:b0:4e:08:27:b8:00:57:35:
        77:da:bd:8c:07:cf:92:35:63:e7:0e:b6:fa:a2:3b:66:1d:28:
        e3:90:66:85:16:e1:78:1a:ba:da:e6:32:73:52:07:1f:6f:54:
        4d:2a:d0:4c:f8:1e:3a:a0:f2:b1:c7:ab:83:b5:aa:b4:24:3a:
        25:16:55:98:20:0e:80:65:aa:e4:52:52:bb:04:a9:dd:6a:04:
        97:ae:7e:87:35:6e:cb:34:79:2c:63:37:fa:d0:e8:a8:11:99:
        ec:8f:d2:70:c1:59:64:b6:ca:dc:8c:5a:ad:e8:20:fe:f9:38:
        bd:01:1a:7c:b9:d0:10:b1:2e:6e:d0:ae:a2:1c:07:a2:2e:90:
        0d:1f:ff:ba:ad:26:54:33:9c:70:5f:32:68:0b:6f:5a:25:a5:
        f4:5f:ce:3c:f1:73:ba:10:af:e2:91:dc:9d:8a:5a:26:2f:a7:
        a4:70:aa:51:96:a7:fc:f6:74:4d:d7:2b:20:1d:c4:af:d3:84:
        6a:65:fd:a5:8f:cd:c0:b1:a7:90:05:f9:c3:86:c0:f9:6c:55:
        7c:64:7e:9b
-----BEGIN CERTIFICATE-----
MIID7jCCAtagAwIBAgIBATANBgkqhkiG9w0BAQsFADAlMSMwIQYDVQQDDBoxMDcu
MTc4LjIwOS4xMDNAMTQzNDczODYzNTAeFw0xNTA2MTkxODMwMzVaFw0yNTA2MTYx
ODMwMzVaMBwxGjAYBgNVBAMMEWt1YmVybmV0ZXMtbWFzdGVyMIIBIjANBgkqhkiG
9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvXWgbf9qqxTrb0kL/1SvriZD8ob8s1cmaiNe
mBQQxUfkHDc5o3ntoNR1bQ6iwZtZTY36qdhQKT3q5AXoZrYour4smDlxGxnrNvjU
unwim6Gf51+2CWOjmWVPXp6pvsKjUgSA0KPvViztHk/JmrdYorqq19LyaaVi7k0H
IOi+xlCC60C32oe0VZ85WcSiBOKSYytAVv4M/eEfCSFN/2l9ZUpbcNGZFSVEJBBt
xlpmiwqaN3AozVJKVMsRhblhIS+h2FfwzZEFwntuGXgnE9pEHMUZtg7NxS4GfLd1
1luFbtU3zw8lobuli8l7l+py3s+eiEFkna5V03Lu9cdXi4AdNwIDAQABo4IBMDCC
ASwwCQYDVR0TBAIwADAdBgNVHQ4EFgQUlgAdTXQqEnbk8ZV0JFzLgnHr7PEwVQYD
VR0jBE4wTIAU7dElKm5we5fhFQLYFnyu+roTfAChKaQnMCUxIzAhBgNVBAMMGjEw
Ny4xNzguMjA5LjEwM0AxNDM0NzM4NjM1ggkA+a8fn3EJOLEwEwYDVR0lBAwwCgYI
KwYBBQUHAwEwCwYDVR0PBAQDAgWgMIGGBgNVHREEfzB9hwRrstFnhwQKAAABggpr
dWJlcm5ldGVzghJrdWJlcm5ldGVzLmRlZmF1bHSCFmt1YmVybmV0ZXMuZGVmYXVs
dC5zdmOCJGt1YmVybmV0ZXMuZGVmYXVsdC5zdmMuY2x1c3Rlci5sb2NhbIIRa3Vi
ZXJuZXRlcy1tYXN0ZXIwDQYJKoZIhvcNAQELBQADggEBALgLh9h7d2XY15mJ3vOe
fSkP9lfbox9JH0I374x5SzjEyyDVyJddrj+c0JQ8kPCwTggnuABXNXfavYwHz5I1
Y+cOtvqiO2YdKOOQZoUW4XgautrmMnNSBx9vVE0q0Ez4Hjqg8rHHq4O1qrQkOiUW
VZggDoBlquRSUrsEqd1qBJeufoc1bss0eSxjN/rQ6KgRmeyP0nDBWWS2ytyMWq3o
IP75OL0BGny50BCxLm7QrqIcB6IukA0f/7qtJlQznHBfMmgLb1olpfRfzjzxc7oQ
r+KR3J2KWiYvp6RwqlGWp/z2dE3XKyAdxK/ThGpl/aWPzcCxp5AF+cOGwPlsVXxk
fps=
-----END CERTIFICATE-----
```

which looks correct to me. I'll run an e2e test locally as well.

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -44,6 +44,8 @@ CLUSTER_IP_RANGE="${CLUSTER_IP_RANGE:-10.244.0.0/16}"
 MINION_SCOPES=("storage-ro" "compute-rw" "https://www.googleapis.com/auth/monitoring" "https://www.googleapis.com/auth/logging.write")
 # Increase the sleep interval value if concerned about API rate limits. 3, in seconds, is the default.
 POLL_SLEEP_INTERVAL=3
+# TODO(roberthbailey): If SERVICE_CLUSTER_IP_RANGE is smaller than a /24 then
```
Comment (thockin) [inline cluster/gce/config-default.sh]: We should enforce this.  can we parse this somewhere (maybe your create-certs func, or maybe earlier) and assert that the mask is <=24?

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -542,8 +542,12 @@ function write-node-env {
 #   KUBECFG_CERT_BASE64
 #   KUBECFG_KEY_BASE64
 function create-certs {
-  local cert_ip
-  cert_ip="${1}"
+  local -r cert_ip="${1}"
+
+  local octects=($(echo "$SERVICE_CLUSTER_IP_RANGE" | sed -e 's|/.*||' -e 's/\./ /g'))
+  octects[3]=1
```
Comment (thockin) [inline cluster/gce/util.sh]: If we change this to `((OCTETS[3]+=1))`, then you handle any mask properly.

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (thockin) [inline cluster/gce/config-default.sh]: if you make the below change (increment) we don't even need this diff chunk

**Comment by thockin** [comment]
LGTM but the increment

File: cluster/gce/util.sh (hunk 2 above)
Comment (roberthbailey) [inline cluster/gce/util.sh]: done.

**Comment by roberthbailey** [comment]
comments addressed.

**Comment by thockin** [comment]
LGTM

Thanks

**Comment by satnam6502** [comment]
This is v1 candidate and LGTMed. Wait for this to be triaged to v1? Or mege anyway?

**Comment by roberthbailey** [comment]
I haven't figured out how to kick shippable, but I don't see how this change could have broken it.

**Comment by satnam6502** [comment]
I've wasted lots of time trying to work out how to re-run this in Shippable and given up.

**Comment by brendandburns** [comment]
For shippable, it's in the overall build list and not the build page.

**Comment by brendandburns** [comment]
(and it appears to be a flake)

**Comment by satnam6502** [comment]
I think I looked there and I could not find this specific build but I could see others that I could re-run.

**Comment by satnam6502** [comment]
Well after doing Fetch More lots and lots of time I managed to re-run Shippable.

**Comment by jlowdermilk** [comment]
Per updated merge policy, please add risk assessment to description. Assigning to @davidopp  for a second lgtm.

**Comment by roberthbailey** [comment]
Risk: low. This change just adds a few new strings to the master's certificate.

**Comment by davidopp** [comment]
LGTM

**Pull request merged**